specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/storageTaskAssignments.json

```diff
@@ -0,0 +1,842 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/storageTaskAssignments": {
+      "get": {
+        "tags": [
+          "StorageTaskAssignments"
+        ],
+        "description": "List all the storage task assignments in an account",
+        "operationId": "StorageTaskAssignments_List",
+        "x-ms-examples": {
+          "ListStorageTaskAssignmentsForAccount": {
+            "$ref": "./examples/storageTaskAssignmentsList/ListStorageTaskAssignmentsForAccount.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./storage.json#/parameters/StorageAccountName"
+          },
+          {
+            "name": "$maxpagesize",
+            "in": "query",
+            "type": "string",
+            "description": "Optional, specifies the maximum number of storage task assignment Ids to be included in the list response."
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "type": "string",
+            "description": "Optional. When specified, it can be used to query using reporting properties."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- list of storage task assignments was retrieved and returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/StorageTaskAssignmentsList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/reports": {
+      "get": {
+        "tags": [
+          "StorageTaskAssignments"
+        ],
+        "description": "Fetch the report summary of all the storage task assignments and instances in an account",
+        "operationId": "StorageTaskAssignmentsInstancesReport_List",
+        "x-ms-examples": {
+          "ListStorageTaskAssignmentsInstancesReportSummary": {
+            "$ref": "./examples/storageTaskAssignmentsList/ListStorageTaskAssignmentsInstancesReportSummary.json"
+          }
+        },
+        "parameters": [
```

Comment:
Why do you have a 'list report resources' API but no corresponding 'point GET' - to get details of just one report resource?

"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/reports"

The report resource, if its a resource, is going to include an 'id' property in the response, which supposed to link to some uri like

'/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/reports/{reportid}'

I presume - but will people actually be able to fetch report details with a GET based on knowing the id - if anyone shares them that link?

Reply:
Oh, the resource id in the response is going to be like this?

            "type": "Microsoft.Storage/storageAccount/{accoutName}/reports/{reportid}"?

I wonder if this works properly with RBAC

Reply:
I think we are having the same loop of questions each time the reviewer for the weke changes Tim :). Unsure when the review would end, its pretty passive with lack of handover looks like , a bit of feedback on the process :)

I think having a call and providing whole context again would help better, let us know if that works for you Tim ?


Answered this here 
https://github.com/Azure/azure-rest-api-specs/pull/28461#discussion_r1573045656
https://github.com/Azure/azure-rest-api-specs/pull/28461#discussion_r1578436948

Reply:
Sorry for the repetition, I'm mainly questioning it because it seems unusual, not impossible. :)